packages/web-components/src/components/masthead/__stories__/masthead.stories.ts

```diff
@@ -87,6 +102,55 @@ export const Default = ({ parameters }) => {
   `;
 };
 
+export const WithCustomTypeahead = ({ parameters }) => {
+  const { customProfileLogin, platform, selectedMenuItem, userStatus, searchPlaceholder, hasProfile, hasSearch } =
+    parameters?.props?.MastheadComposite ?? {};
+
+  document.documentElement.addEventListener('dds-search-with-typeahead-input', async e => {
+    const results = await customTypeaheadApiFunction((e as CustomEvent).detail.value);
+    document.dispatchEvent(new CustomEvent('dds-custom-typeahead-api-results', { detail: results }));
+  });
+
+  return html`
+    <style>
+      ${styles}
+    </style>
+    <dds-masthead-composite
+      platform="${ifNonNull(platform)}"
+      .platformUrl="${ifNonNull(platformData.url)}"
+      selected-menu-item="${ifNonNull(selectedMenuItem)}"
+      user-status="${ifNonNull(userStatus)}"
+      searchPlaceholder="${ifNonNull(searchPlaceholder)}"
+      .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
```

Comment:
Looks great! Just one minor thing. Can we add `navLinks` to the story
<img width="787" alt="Screen Shot 2022-01-24 at 12 06 15 PM" src="https://user-images.githubusercontent.com/20210594/150856442-40f7f917-1a04-4dc0-9890-66c6268798b3.png">



```suggestion
      .navLinks="${navLinks}"
      .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
```